prime-router/src/test/kotlin/fhirengine/azure/FhirFunctionTests.kt

```diff
@@ -732,30 +802,162 @@ class FhirFunctionTests {
                 queueAccess = queueMock,
                 databaseAccess = ReportStreamTestDatabaseContainer.testDatabaseAccess
             )
+
             fhirFunc.doTranslate(queueMessage, 1, fhirEngine, actionHistory)
 
-            val convertTask = ReportStreamTestDatabaseContainer.testDatabaseAccess.fetchTask(report.id)
-            assertThat(convertTask.translatedAt).isNotNull()
+            // verify task and report_file tables were updated correctly in the Translate function (new task and new
+            // record file created)
             ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
-                val routeTask = DSL.using(txn).select(Task.TASK.asterisk()).from(Task.TASK)
+                val queueTask = DSL.using(txn).select(Task.TASK.asterisk()).from(Task.TASK)
                     .where(Task.TASK.NEXT_ACTION.eq(TaskAction.batch))
                     .fetchOneInto(Task.TASK)
-                assertThat(routeTask).isNotNull()
-                val convertReportFile =
+                assertThat(queueTask).isNotNull()
+
+                val sendReportFile =
                     DSL.using(txn).select(REPORT_FILE.asterisk())
                         .from(REPORT_FILE)
-                        .where(REPORT_FILE.NEXT_ACTION.eq(TaskAction.batch))
+                        .where(REPORT_FILE.REPORT_ID.eq(queueTask!!.reportId))
                         .fetchOneInto(REPORT_FILE)
-                assertThat(convertReportFile).isNotNull()
+                assertThat(sendReportFile).isNotNull()
+            }
+            // verify we DID NOT download the original report in the Translate function
+            verify(exactly = 0) {
+                BlobAccess.Companion.downloadBlobAsByteArray(convertReport.bodyURL, any(), any())
+            }
+            // verify we called the logic to translate the bodyURL passed to translate function
+            verify(exactly = 1) {
+                fhirEngine.getByteArrayFromBundle(any(), any())
             }
+            // verify sendMessage did not get called because next action should be Batch and the batch decider function,
+            // which runs on a timer, will pick up the task from the DB instead of being called directly by the
+            // Translate function
             verify(exactly = 0) {
                 queueMock.sendMessage(any(), any())
             }
         }
 
+        @Test
+        fun `test successfully processes a translate message when isSendOriginal is true`() {
```

Comment:
Nit: maybe also one checking that HL7 works as well?

Reply:
Will revisit in #13232